app/controllers/discover_controller.rb

```diff
@@ -134,4 +138,8 @@ def prepare_discover_page
         @discover_tag_meta_description = presenter.meta_description
       end
     end
+
+    def black_friday_feature_active?
+      Feature.active?(:offer_codes_search) || (params[:feature_key].present? && ActiveSupport::SecurityUtils.secure_compare(params[:feature_key].to_s, ENV["SECRET_FEATURE_KEY"].to_s))
```

Comment:
@EmCousin we can look into enabling Feature flag for the logged-in user in a new PR, for testing purposes.

Reply:
PR here: https://github.com/antiwork/gumroad/pull/2140